src/tiered/tiered_handle.c

```diff
@@ -19,27 +19,52 @@ __tiered_name_check(WT_SESSION_IMPL *session, WT_TIERED *tiered)
 {
     WT_DECL_RET;
     WT_FILE_SYSTEM *bucket_fs;
+    size_t len, obj_len;
     u_int obj_count, i;
     char **obj_files;
-    const char *name;
+    const char *name, *obj_name, *obj_uri;
 
     bucket_fs = tiered->bstorage->file_system;
     name = tiered->iface.name;
+    obj_name = obj_uri = NULL;
     WT_ASSERT(session, WT_PREFIX_MATCH(name, "tiered:"));
     WT_PREFIX_SKIP(name, "tiered:");
     /* See if this name exists in the shared storage. */
     __wt_verbose(session, WT_VERB_TIERED, "NAME_CHECK: check for %s", name);
-    WT_ERR(bucket_fs->fs_directory_list(
+    WT_RET(bucket_fs->fs_directory_list(
       bucket_fs, (WT_SESSION *)session, NULL, name, &obj_files, &obj_count));
     __wt_verbose(session, WT_VERB_TIERED, "NAME_CHECK: Got %d files", (int)obj_count);
-    for (i = 0; i < obj_count; ++i)
+    if (obj_count == 0)
+        goto done;
+    /*
+     * We need to handle overlapping naming. Generate an object name so that we know the maximum
```

Comment:
I'm concerned that a reader might to understand the issue here just from seeing the phrase "overlapping naming."  Maybe expand this to say something about distinguishing a case where other table name is a superstring or the one we're dropping?

Reply:
Changed.